src/rts/GameState.java

```diff
@@ -344,24 +344,32 @@ public boolean issueSafe(PlayerAction pa) {
             }
             
             // get the unit that corresponds to that action (since the state might have been cloned):
-            if (pgs.units.indexOf(p.m_a)==-1) {
-                boolean found = false;
-                for(Unit u:pgs.units) {
-                    if (u.getClass()==p.m_a.getClass() &&
-//                        u.getID() == p.m_a.getID()) {
-                        u.getX()==p.m_a.getX() &&
-                        u.getY()==p.m_a.getY()) {
-                        p.m_a = u;
-                        found = true;
-                        break;
-                    }
-                }
-                if (!found) {
-                    System.err.println("Inconsistent order: " + pa);
+            boolean foundRealUnit = false;
+            Unit substituteUnit = null;
+            for (final Unit u : pgs.units) {
+            	if (u.equals(p.m_a)) {
+            		foundRealUnit = true;
+            		break;
+            	}
+            	
+            	if (substituteUnit == null) {
+            		// TODO should we also compare u.getType() to p.m_a.getType()?
```

Comment:
the original code had a `u.getClass() == p.m_a.getClass()` check, which I removed because it was useless (there are no subclasses of `Unit`). Maybe the intention was to actually check for equal types though? If so, I could add that.